docs/encyclopedia/application-message-passing.mdx

```diff
@@ -220,22 +220,40 @@ Every time the Workflow wakes up--generally, it wakes up when it needs to--it wi
 
 This execution is on a single thread–while this means you don’t have to worry about parallelism, you do need to worry about concurrency if you have written Signal and Update handlers that can block. These can run interleaved with the main Workflow and with one another, resulting in potential race conditions. These methods should be made reentrant.
 
+#### Initializing the Workflow first {#workflow-initializers}
+
+Initialize your Workflow's state before handling messages. This prevents reading uninitialized instance variables.
+
+To see why, refer to the [diagram](#message-handler-concurrency). It shows that your Workflow processes messages before the first run of your Workflow's main method.
+
+This happens in several scenarios, such as:
```

Comment:
Suggestion: be more specific what "this" is. Specifically the early execution of message handlers. Also be clear whether it "happens" or "might happen".